### PR TITLE
ISSUE-326 Expose request timeout configuration for retrieving datafile

### LINKED ISSUE #326: The ability to customise request timeout in ConfigManager
The default settings for 
- Blocking request timeout
- Polling interval
- Request timeout
are all default to 10 seconds

Whilst the library offers the ability to override the Blocking and Polling settings, the request timeout is not exposed. 
 Can you confirm that this is a design decision, or would you be happy to accept our PR to expose the timeout setting also?

The reason this is important to us is because of the need to limit and have predictable message processing duration in our high-throughput system, and to de-risk ourselves from any network problem.

We would be happy to accept the risk that if we will fall back to the default variant on timeout, it can cause a discrepancy in the recorded metrics of the fallback variant being used instead of the variant that may be returned eventually; we expect this the occurrence should be rare.

### PR DESCRIPTION
**Pull request by GordonSo**
Summary
-------

-  Offer the ability to customise request timeout in ConfigManager.

To have a more predictable duration on experiment variant retrieval duration.

Test plan
---------
Same mimic'd test from blocking_timeout

Fixes #326

### CODE REVIEW

**Comment by GordonSo** [comment]
The travis CI build is only recognising branch against the origin repo (which I have no access to); if somebody with great power push it into under branch and open a PR from it that should 'fix' the pipeline. 

NB: cliche might be sound, but the tests are passing on my machine 😆 
image

File: optimizely/config_manager.py [hunk 1]
```diff
@@ -198,6 +199,7 @@ def __init__(
         )
         self.set_update_interval(update_interval)
         self.set_blocking_timeout(blocking_timeout)
+        self.set_request_timeout(request_timeout)
```
[the remark this answers is not in the document]
Comment (GordonSo) [reply optimizely/config_manager.py:204]: Thanks for the comments, REQUEST_TIMEOUT renamed was done via refactoring into DEAFAULT_REQUEST_TIMEOUT and I have double-checked for any reminding presence with "find all".

docstring is now updated.

**Comment by GordonSo** [comment]
> @GordonSo Thanks for your contribution. We'll take a look at it.
> Can you sign our CLA if not done yet: https://github.com/optimizely/swift-sdk/blob/master/CONTRIBUTING.md

Cheers, I have signed.

**Comment by GordonSo** [comment]
Please let me know if there is anything else 🙏

**Comment by jaeopt** [comment]
@GordonSo Thanks for clarification. 
I think what you want is to set "blocking_timeout" to your tighter limit. 

The Optimizely SDK won't hold CPU long (~msecs) once datafile fetch is completed. When network connection fails, if you set "blocking_timeout" less than the default "request_timeout", then decide API will return fallback on "blocking_timeout".  Datafile fetching will continue in a thread, and dropped at the request timeout or update config successfully.

**Comment by GordonSo** [comment]
> @GordonSo Thanks for clarification.
> I think what you want is to set "blocking_timeout" to your tighter limit.
> 
> The Optimizely SDK won't hold CPU long (~msecs) once datafile fetch is completed. When network connection fails, if you set "blocking_timeout" less than the default "request_timeout", then decide API will return fallback on "blocking_timeout". Datafile fetching will continue in a thread, and dropped at the request timeout or update config successfully.

Thank you!  We shall go with that existing parameter under load test and will revisit if we counter any problems.

**Pull request closed without merging**